### PR TITLE
Add MockAddon for testing

### PR DESCRIPTION
**Pull request by bewee**

### CODE REVIEW

**Comment by bewee** [comment]
Should we move the mock addon into a separate repository @tim-hellhake ?

File: mock-addon/build.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  linux-x64)
+    install_linux_cross_compiler
+    build_cross_compiled
+    ;;
+
+  *)
+    echo "Unsupported architecture"
+    exit 1
+    ;;
+esac
```
Comment (tim-hellhake) [inline mock-addon/build.sh:82]: We don't need this

File: mock-addon/package.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+mkdir -p package
+cp "${FILES[@]}" package
+
+cd package
+shasum --algorithm 256 * > SHA256SUMS
+cd ..
+
+tar -czvf "${TARFILE}" package/*
+shasum --algorithm 256 "${TARFILE}" > "${TARFILE}".sha256sum
```
Comment (tim-hellhake) [inline mock-addon/package.sh:29]: We don't need this

**Review by tim-hellhake — APPROVED** [review]

**Comment by tim-hellhake** [comment]
> Should we move the mock addon into a separate repository @tim-hellhake ?

It's not very useful outside the gateway tests so I don't think it's worth it.

**Comment by bewee** [comment]
Finally 🥳

**Pull request merged**